src/tests/test.ts

```diff
@@ -185,24 +185,29 @@ test('device.createSendTransport() for sending media succeeds', () =>
 		sctpParameters
 	} = fakeParameters.generateTransportRemoteParameters();
 
-	sendTransport = device.createSendTransport(
+	sendTransport = device.createSendTransport<{ foo: number }>(
```

Comment:
Your issue is that above you've already defined `sendTransport` as a `mediasoupClient.types.Transport`, which has a default type for `AppData` of `{ [key: string]: unknown }`

You can see what I mean just by putting a `const` at the beginning of line 188 so the compiler sees this as a fresh binding.

Reply:
OMG so you mean that it's just an issue in the tests file because I'm reusing a let variable???